### PR TITLE
Persist UI changes in config.toml

### PR DESCRIPTION
**Pull request by tadaskay**
`GET /config/user` returns current user configuration
`POST /config/user` changes and saves user configuration

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	assert.Equal(t, 40000, cfg.GetInt("openvpn.port"))
+
+	// when: user configuration is saved
+	err = cfg.SaveUserConfig()
+	// then: only user configuration values are stored
+	assert.NoError(t, err)
+	tomlContent, err := ioutil.ReadFile(configFileName)
+	assert.NoError(t, err)
+	assert.Contains(t, string(tomlContent), "port = 22822")
+	assert.NotContains(t, string(tomlContent), "proto = \"tcp\"")
```
Comment (vkuznecovas) [inline config/config_test.go]:
```suggestion
	assert.NotContains(t, string(tomlContent), `proto = "tcp"`)
```

File: config/urfavecli/clicontext/user.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		return err
+	}
+
+	return config.Current.LoadUserConfig(configFilePath)
+}
+
+// LoadUserConfigQuietly like LoadUserConfig, but instead of returning an error,
+// it logs it on a `warn` level.
+// `error` is specified as a return to adhere to `cli.BeforeFunc` for convenience.
+func LoadUserConfigQuietly(ctx *cli.Context) error {
```
Comment (vkuznecovas) [inline config/urfavecli/clicontext/user.go]: don't need the return value here

File: config/urfavecli/clicontext/user.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	return nil
+}
+
+func dirExists(dir string) error {
+	fileStat, err := os.Stat(dir)
+	if err != nil {
+		return err
+	}
+	if isDir := fileStat.IsDir(); !isDir {
+		return errors.New("directory expected")
```
Comment (vkuznecovas) [inline config/urfavecli/clicontext/user.go]: would be nice to specify the dir in the error

File: utils/jsonutil/strings.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package jsonutil
+
+import "encoding/json"
+
+// ToJson marshals v to a JSON string
+func ToJson(v interface{}) string {
+	bytes, _ := json.Marshal(&v)
```
Comment (vkuznecovas) [inline utils/jsonutil/strings.go]: Would be nice to at least log it here and not ignore it. What if I pass a channel here? :)

File: config/urfavecli/clicontext/user.go (hunk 2 above)
Comment (tadaskay) [reply config/urfavecli/clicontext/user.go]: Need it to implement `BeforeFunc` type from `urfave/cli`:

```go
type BeforeFunc func(*Context) error
```

This is to simplified usage:

```go
return &cli.Command{
		Name:   cliCommandName,
		Usage:  "Starts a CLI client with a Tequilapi",
		Before: clicontext.LoadUserConfigQuietly,
		...
}
```

File: config/urfavecli/clicontext/user.go (hunk 3 above)
Comment (tadaskay) [reply config/urfavecli/clicontext/user.go]: Point taken

File: utils/jsonutil/strings.go (hunk 4 above)
Comment (tadaskay) [reply utils/jsonutil/strings.go]: Point taken

File: utils/jsonutil/strings.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/mysteriumnetwork/node/logconfig"
+)
+
+var log = logconfig.NewLogger()
+
+// ToJson marshals v to a JSON string
+func ToJson(v interface{}) string {
+	bytes, err := json.Marshal(&v)
+	log.Warnf("failed to marshal to JSON string: %v", err)
```
Comment (soffokl) [inline utils/jsonutil/strings.go]: We probably lacking if block here to check the `err`.

Comment (tadaskay) [reply utils/jsonutil/strings.go]: This is intended. I'd like it to use in simple one-line situations, where error checking is not mandatory, e.g.:

```go
log.Info("user configuration loaded: \n", jsonutil.ToJson(cfg))
```

Comment (soffokl) [reply utils/jsonutil/strings.go]: But you are logging every time record `failed to marshal to JSON string` even if no error happened. Or I missing something?

Comment (tadaskay) [reply utils/jsonutil/strings.go]: Ah I misunderstood you. Thanks, changed it. :)

**Pull request merged**